### PR TITLE
v2: fix squished location buttons and add coordinate sign toggle

### LINKED ISSUE #1021: UI Fit and Finish
### Summary

1.)  The keyboard for manually setting location is missing - and + signs (first screen shot).  I suspect the buttons on either side of the zero were meant to be such, but on my Samsung Galaxy S23 Android phone they are inoperable.  I was able to paste the - sign into the longitude from the clipboard.

2.)  The "Change Location" button is squished (second screen shot).

### Steps to Reproduce

_No response_

### Expected Behavior

_No response_

### Observed Behavior

_No response_

### Device

_No response_

### Android Version

_No response_

### Sky Map Version

2.x

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by jaydeetay**
## Summary
- Fix the squished "Change Location" button by switching `LocationSheet`'s button row from `Row` to `FlowRow`, matching the fix already applied to `TimeTravelDialog` for the same issue.
- Apply the same `Row` → `FlowRow` fix to the EULA screen's Accept/Decline button bar, found during a sweep for the same pattern.
- Add an in-app +/- toggle button to the latitude/longitude fields in the manual location entry dialog, since some OEM decimal keypads (e.g. Samsung) render +/- keys that don't actually commit a sign.

Fixes #1021.

## Test plan
- [ ] `./gradlew check` (ktlint, unit tests, Konsist) passes in stardroid-v2
- [ ] Manually verify the location sheet buttons wrap instead of clipping on a narrow/small screen
- [ ] Manually verify the EULA Accept/Decline buttons wrap correctly
- [ ] Manually verify the +/- toggle on the latitude/longitude fields lets you enter a negative coordinate

🤖 Generated with [Claude Code](https://claude.ai/code)

___

### **PR Type**
Bug fix

___

### **Description**
- Wrap location action buttons using `FlowRow`

- Add sign toggle to coordinate entry fields

- Wrap EULA action buttons using `FlowRow`

___

### Diagram Walkthrough

```mermaid
flowchart LR
  subgraph LocationUI ["LocationUi.kt"]
    direction TB
    FlowLoc["FlowRow for location buttons"]
    SignToggle["SignToggle on coordinate text fields"]
  end
  subgraph StartupDialogs ["StartupDialogs.kt"]
    FlowEula["FlowRow for EULA buttons"]
  end
  subgraph Resources ["strings.xml"]
    ToggleString["location_toggle_sign string"]
  end

  ToggleString -- "content description" --> SignToggle
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Bug fix</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>LocationUi.kt</strong><dd><code>Wrap location buttons and add coordinate sign toggle</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

stardroid-v2/app/src/main/kotlin/com/google/android/stardroid/ui/location/LocationUi.kt

<ul><li>Replaced <code>Row</code> with <code>FlowRow</code> for location action buttons to prevent <br>squishing on narrow screens.<br> <li> Introduced <code>SignToggle</code> composable to toggle coordinate positive and <br>negative signs.<br> <li> Added <code>SignToggle</code> as trailing icons in latitude and longitude entry <br>fields.</ul>

</details>

  </td>
  <td><a href="https://github.com/sky-map-team/stardroid/pull/1023/files#diff-ee472ac77d4e59e58411ceeb177f595196ffa5634cd337729e8646c1e4dded3b">+40/-2</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>StartupDialogs.kt</strong><dd><code>Wrap EULA bottom bar buttons with FlowRow</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

stardroid-v2/app/src/main/kotlin/com/google/android/stardroid/ui/startup/StartupDialogs.kt

<ul><li>Switched the EULA bottom bar buttons from <code>Row</code> to <code>FlowRow</code> to prevent <br>clipping across long translations and narrow screens.</ul>

</details>

  </td>
  <td><a href="https://github.com/sky-map-team/stardroid/pull/1023/files#diff-2bcc2cdce1f107b72c8151fba5ddb513b8111e58b0c58602e74e2f63f154a41a">+8/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>strings.xml</strong><dd><code>Add accessibility string for sign toggle button</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

stardroid-v2/app/src/main/res/values/strings.xml

<ul><li>Added <code>location_toggle_sign</code> string resource for accessibility on the <br>coordinate sign toggle button.</ul>

</details>

  </td>
  <td><a href="https://github.com/sky-map-team/stardroid/pull/1023/files#diff-47ca36ffdf2cb8ecdfa60f7be7d0fd8eb0af8143da77ba0c206d52b401243934">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

</details>

___

### CODE REVIEW

**Comment by jaydeetay** [comment]
@claude Please review this PR for correctness, threading safety, and adherence to project patterns. Check AGENTS.md and CLAUDE.md for conventions. Focus on:
1. Correctness and thread safety (data races, concurrent modification, null safety)
2. API contracts and backward compatibility
3. Architecture and design patterns (Hilt injection, layer separation, rendering pipeline)
4. Performance concerns (allocations, unnecessary copying, rendering overhead)
5. Security issues (input validation, access control, sensitive data handling)

Provide line-level feedback with specific suggestions.

File: stardroid-v2/app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -264,6 +264,7 @@
     <string name="location_geocoder_offline" translation_description="Inline message shown when geocoding fails because there is no internet connection">No internet — enter coordinates directly</string>
     <string name="location_invalid_latitude" translation_description="Validation error for latitude values outside the valid range">Latitude must be between −90 and 90</string>
     <string name="location_invalid_longitude" translation_description="Validation error for longitude values outside the valid range">Longitude must be between −180 and 180</string>
+    <string name="location_toggle_sign" translation_description="Accessibility label for a button that flips a coordinate field between positive and negative">Toggle sign</string>
```
Comment (jaydeetay) [inline stardroid-v2/app/src/main/res/values/strings.xml]: THis seems too verbose.

**Comment by jaydeetay** [comment]
@claude Addressing your review:

1. Fixed in ba53c0a3: `SignToggle`'s content description now reads "Make positive"/"Make negative" based on current sign, replacing the static "Toggle sign" string.
2. Deferring: a bare `-` on an empty field is acceptable placeholder text pending digit entry — not worth extra state-hiding logic for this edge case.
3. Deferring: covered by the PR's existing manual-test plan (verifying the EULA button row wraps without clipping); no code change needed unless that check finds an actual bug.
4. No action: as you noted, self-consistent given the toggle only ever writes ASCII `-`; not a real input path today.

**Comment by jaydeetay** [comment]
@claude Please review the latest changes.

File: stardroid-v2/app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -264,6 +264,8 @@
     <string name="location_geocoder_offline" translation_description="Inline message shown when geocoding fails because there is no internet connection">No internet — enter coordinates directly</string>
     <string name="location_invalid_latitude" translation_description="Validation error for latitude values outside the valid range">Latitude must be between −90 and 90</string>
     <string name="location_invalid_longitude" translation_description="Validation error for longitude values outside the valid range">Longitude must be between −180 and 180</string>
+    <string name="location_toggle_sign_make_positive" translation_description="Accessibility label for a button that flips a currently-negative coordinate field to positive">Make positive</string>
```
Comment (jaydeetay) [inline stardroid-v2/app/src/main/res/values/strings.xml:267]: This still seems too verbose.

**Comment by jaydeetay** [comment]
@claude to be clear I'm not objecting to the length of the description - I'm just concerned about a button with "Make positive" on it being quite big for a feature that only a few users need.

File: stardroid-v2/app/src/main/res/values/strings.xml (hunk 2 above)
Comment (jaydeetay) [reply stardroid-v2/app/src/main/res/values/strings.xml:267]: Fixed in 4cfc02c3: shortened both descriptions further, e.g. "Accessibility label for the button that switches this coordinate to positive".

File: stardroid-v2/app/src/main/res/values/strings.xml (hunk 1 above)
Comment (jaydeetay) [reply stardroid-v2/app/src/main/res/values/strings.xml]: Fixed in 4cfc02c3: trimmed both translation_description attributes to drop the redundant "currently-X coordinate field" phrasing.

**Comment by jaydeetay** [comment]
@claude Please review the latest changes.

**Comment by jaydeetay** [comment]
Screenshot_20260914-211227

Hmm. I'm not sure this works.  It doesn't look intuitive to me.
